addons/core/translations/resources/en-us.yaml

```diff
@@ -88,6 +89,15 @@ host-catalog:
     unknown: Unknown
     aws: AWS
     azure: Azure
+    dynamic: Dynamic
+  form:
+    provider-details: 
```

Comment:
If we are reflecting the existing pattern, this would be a a form field name that is underscored rather than dasherized.